src/main/java/uk/ac/sanger/sccp/stan/service/WorkProgressServiceImp.java

```diff
@@ -80,26 +93,47 @@ public List<WorkProgress> getProgress(String workNumber, List<String> workTypeNa
             if (statuses!=null && !statuses.contains(singleWork.getStatus())) {
                 return List.of();
             }
+            if (programs!=null && !programs.contains(singleWork.getProgram())) {
+                return List.of();
+            }
             return List.of(getProgressForWork(singleWork, opTypeFilter, stainTypeFilter, labwareTypeFilter,
                     releaseLabwareTypeFilter, labwareIdToLabware, labwareTypeToStainMap));
         }
+
         if (workTypes!=null) {
             List<Work> works = workRepo.findAllByWorkTypeIn(workTypes);
             Stream<Work> workStream = works.stream();
             if (statuses!=null) {
                 workStream = workStream.filter(work -> statuses.contains(work.getStatus()));
             }
+            if (programs!=null) {
+                workStream = workStream.filter(work -> programs.contains(work.getProgram()));
+            }
```

Comment:
Why we are filtering the search using current statuses and programs ? Is this to remove any records having invalid status and programs?

Reply:
In this case the search has specified by work types, so we look up the works by work type; then if the search has *also* specified statuses and/or programs, then we filter out the works that we found by those additional characteristics. The API supports combined searches, even if the client does not.